builtin/providers/aws/resource_aws_lambda_function.go

```diff
@@ -487,3 +541,18 @@ func validateVPCConfig(v interface{}) (map[string]interface{}, error) {
 
 	return config, nil
 }
+
+func validateEnvironment(v interface{}) (map[string]interface{}, error) {
+	environments := v.([]interface{})
+
+	if len(environments) > 1 {
+		return nil, fmt.Errorf("At most one Environment is allowed per Lambda Function")
+	}
+
+	environment, ok := environments[0].(map[string]interface{})
+	if !ok {
+		return nil, errors.New("environment is <nil>")
```

Comment:
I would suggest more meaningful error message - a human being in on the receiving end of it 😄